waspc/packages/deploy/src/providers/fly/helpers/flyctlHelpers.ts

```diff
@@ -0,0 +1,62 @@
+import { exit } from 'process'
+import { $, question } from 'zx'
+import { isYes, waspSays } from './helpers.js'
+
+export async function flyctlExists(): Promise<boolean> {
+  try {
+    await $`flyctl version`
+    return true
+  } catch {
+    return false
+  }
+}
+
+export async function isUserLoggedIn(): Promise<boolean> {
+  try {
+    await $`flyctl auth whoami`
+    return true
+  } catch {
+    return false
+  }
+}
+
+export async function ensureUserLoggedIn() {
+  const userLoggedIn = await isUserLoggedIn()
+  if (!userLoggedIn) {
+    const answer = await question('flyctl is not logged into Fly.io. Would you like to log in now? ')
+    if (isYes(answer)) {
+      try {
+        await $`flyctl auth login`
+      } catch {
+        waspSays(`It seems there was a problem logging in. Please run "flyctl auth login" and try again.`)
+        exit(1)
+      }
+    } else {
+      waspSays(`Ok, exiting.`)
+      exit(1)
+    }
+  }
+}
+
+export async function ensureFlyReady() {
+  if (!await flyctlExists()) {
+    waspSays(`The Fly.io CLI is not available on this system.`)
+    waspSays(`Please install the flyctl here: https://fly.io/docs/hands-on/install-flyctl`)
+    exit(1)
+  }
+  await ensureUserLoggedIn()
+}
+
+export async function ensureRegionIsValid(region: string) {
```

Comment:
Make sure `flyctlExists` runs before this and every other hook. Ref: https://www.notion.so/wasp-lang/Fly-Deploy-comments-61de123caf2348aabb2ccd6d972c84ff